examples/keras_recipes/packaging_keras_models_for_wide_distribution.py

```diff
@@ -0,0 +1,315 @@
+"""
+Title: Packaging Keras models for wide distribution using Functional Subclassing
+Author: Martin Görner
+Date created: 2023-12-13
+Last modified: 2023-12-13
+Description: When sharing your deep learning models, package them using the Functional Subclassing pattern.
+Accelerator: GPU
+"""
+
+"""
+#Introduction
+
+Keras is the ideal framework for sharing your cutting-edge deep learning models, in a
+library of pre-trained (or not) models. Millions of ML engineers are fluent in the
+familiar Keras API, making your models  accessible to a global community, whatever their
+preferred backend (Jax, PyTorch or TensorFlow).
+
+One of the benefits of the Keras API is that it lets users programmatically inspect or
+edit a model, a feature that is necessary when creating new architectures or workflows
+based on a pre-trained model.
+
+When distributing models, the Keras team recommends packaging them using the **Functional
+Subclassing** pattern. Models implemented in this way combine two benefits:
+
+* They can be instantiated in the normal pythonic way:<br/>
+`model = model_collection_xyz.AmazingModel()`
+
+* They are Keras functional models which means that they have a programmatically
+accessible graph of layers, for introspection or model surgery.
+
+This guide explains [how to use](#functional-subclassing-model) the Functional
+Subclassing pattern, and showcases its benefits for [programmatic model
+introspection](#model-introspection) and [model surgery](#model-surgery). It also shows
+two other best practices for sharable Keras models: [configuring
+models](#unconstrained-inputs) for the widest range of supported inputs, for example
+images of various sizes, and [using dictionary inputs](#model-with-dictionary-inputs) for
+clarity in more complex models.
+"""
+
+"""
+# Initialization
+
+The backend must be selected before keras 3 is first imported.
+"""
+
+import keras
```

Comment:
Instead of having per-cell imports, add a "setup" cell at the beginning with the imports and global constants

Reply:
done
renamed "Initialization" to "Setup"